src/test/java/com/artipie/FilesRepoITCase.java

```diff
@@ -81,29 +93,18 @@
      */
     private int port;
 
-    @BeforeEach
-    void init() throws Exception {
-        this.storage = new FileStorage(this.tmp);
-        this.server = new ArtipieServer(this.tmp, "my-file", this.config());
-        this.port = this.server.start();
-        this.server.start();
-        Testcontainers.exposeHostPorts(this.port);
-        this.cntn = new GenericContainer<>("centos:centos8")
-            .withCommand("tail", "-f", "/dev/null")
-            .withWorkingDirectory("/home/")
-            .withFileSystemBind(this.tmp.toString(), "/home");
-        this.cntn.start();
-        this.exec("yum", "-y", "install", "curl");
-    }
-
-    @Test
-    void curlGetShouldReceiveFile() throws Exception {
-        final String url = "http://host.testcontainers.internal:%d/my-file/file-repo/curl.txt";
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void curlGetShouldReceiveFile(final boolean anonymous) throws Exception {
+        this.init(this.config(anonymous));
         this.addFilesToStorage(
             "file-repo", new Key.From("repos", "my-file", "file-repo")
         );
         MatcherAssert.assertThat(
-            this.exec("curl", "-i", "-X", "GET", String.format(url, this.port)),
+            this.exec(
+                "curl", "-i", "-X", "GET", String.format(FilesRepoITCase.URL, this.port),
+                this.flag(anonymous), this.user(anonymous, ArtipieServer.ALICE)
+            ),
```

Comment:
@genryxy methods `flag` and `user` are tightly coupled, they are used together in same way with same argument all the time. I'd suggest to refactor `exec` method to accept `anonymous` flag and add necessary args to the command when enabled.

Maybe we should even rename `exec` to `curl` and omit first string argument, as it is always `"curl"` in this test class.